crates/crypto/bls/src/lib.rs

```diff
@@ -677,4 +677,23 @@ mod tests {
             "The deserialized point does not match the original"
         );
     }
+
+    #[test]
+    fn test_compliance() {
+        let message = "Hello, world!Hello, world!123456";
```

Comment:
is it possible to add a test for the payload which has less than 32 bytes and make sure all is correct? that was what the broken case was

Reply:
The type system won't allow to test an input of less than 32 bytes long with the changes proposed on this PR.

Reply:
oh that's cool. makes sense.